### PR TITLE
Fix things

### PR DESCRIPTION
**Pull request by sadym-chromium**

### CODE REVIEW

File: src/bidiMapper/domains/context/browsingContextImpl.ts [hunk 1]
```diff
@@ -537,13 +537,15 @@ export class BrowsingContextImpl {
     const result = await this.#cdpTarget.cdpClient.sendCommand(
       'Page.addScriptToEvaluateOnNewDocument',
       {
-        source: params.expression,
+        source: `(${params.expression})()`,
```
Comment (sadym-chromium) [inline src/bidiMapper/domains/context/browsingContextImpl.ts:540]: At this point you will have to add a tricks to provide the bindings.

File: src/protocol/protocol.ts [hunk 2]
```diff
@@ -697,7 +697,9 @@ export namespace Script {
   };
 
   export type AddPreloadScriptResult = {
-    script: PreloadScript;
+    result: {
+      script: PreloadScript;
+    };
```
Comment (thiagowfx) [inline src/protocol/protocol.ts:702]:
```
script.AddPreloadScriptResult = {
  script: script.PreloadScript
}
```

```
script.PreloadScript = text;
```

`result` is not in the spec

File: tests/test_preload_script.py [hunk 3]
```diff
@@ -30,9 +31,17 @@ async def test_addPreloadScript_setGlobalVariable(websocket, context_id):
             }
         })
 
-    resp = await read_JSON_message(websocket)
+    assert result == {'script': ANY_STR}
 
-    assert resp == {'id': 1, 'script': '1'}
+    await execute_command(
```
Comment (sadym-chromium) [inline tests/test_preload_script.py:36]: The new script should be run only on the new frames. So navigation is needed.

File: tests/test_preload_script.py [hunk 4]
```diff
@@ -47,31 +56,45 @@ async def test_addPreloadScript_setGlobalVariable(websocket, context_id):
...
+            "method": "browsingContext.navigate",
+            "params": {
+                "url": "data:text/html,<h2>test</h2>",
+                "wait": "complete",
+                "context": context_id
+            }
+        })
 
-    assert resp == {'id': 1, 'script': '1'}
+    resp = await read_JSON_message(websocket)
```
Comment (sadym-chromium) [inline tests/test_preload_script.py:87]: Assert event is happening first

File: tests/test_preload_script.py [hunk 5]
```diff
@@ -47,31 +56,45 @@ async def test_addPreloadScript_setGlobalVariable(websocket, context_id):
...
+                "type": "string",
+                "value": "my preload script"
+            }]
+        }
+    })
 
-    await wait_for_event(websocket, "log.entryAdded")
+    # Assert navigation is finished.
+    resp = await read_JSON_message(websocket)
+    assert resp == AnyExtending({"id": command_id, "result": ANY_DICT})
```
Comment (sadym-chromium) [inline tests/test_preload_script.py:100]: And navigation is finished afterwards.

File: src/protocol/protocol.ts (hunk 2 above)
Comment (sadym-chromium) [reply src/protocol/protocol.ts:702]: {
  id: js-uint,
  result: script.AddPreloadScriptResult,
}

File: src/bidiMapper/domains/context/browsingContextImpl.ts [hunk 6]
```diff
@@ -537,13 +537,15 @@ export class BrowsingContextImpl {
...
         worldName: params.sandbox,
       }
     );
 
     return {
-      script: result.identifier,
+      result: {
```
[the remark this answers is not in the document]
Comment (sadym-chromium) [reply src/bidiMapper/domains/context/browsingContextImpl.ts:546]:
```
CommandResponse: {
  id: js-uint,
  result: script.AddPreloadScriptResult,
}
```

**Comment by sadym-chromium** [comment]
**E2E tests / e2e** Successful

File: src/protocol/protocol.ts (hunk 2 above)
Comment (sadym-chromium) [reply src/protocol/protocol.ts:702]: It looks like the return types need some refactoring, as they are quite confusing.

File: src/bidiMapper/domains/context/browsingContextImpl.ts (hunk 1 above)
Comment (thiagowfx) [reply src/bidiMapper/domains/context/browsingContextImpl.ts:540]: Ack

File: tests/test_preload_script.py (hunk 3 above)
Comment (thiagowfx) [reply tests/test_preload_script.py:36]: Ack. Resolved.

File: src/protocol/protocol.ts (hunk 2 above)
Comment (thiagowfx) [reply src/protocol/protocol.ts:702]: Agreed. I can do so in a follow-up CL.

**Pull request merged**